src/hdmf/data_utils.py

```diff
@@ -241,7 +242,7 @@ def __init__(self, **kwargs):
                 for buffer_axis, maxshape_axis in zip(self.buffer_shape, self.maxshape)
             ],
         )
-        self.buffer_selection_generator = (
+        self.buffer_selections = tuple(  # temporary solution to cast as tuple to allow indexing
```

Comment:
This may be able to be reverted back to the previous state depending on how the actual selection iteration is performed on the Zarr side (TBD)

Reply:
Yep, no longer needed